### DISCUSSION TITLE
upload media file in django

### DISCUSSION THREAD

**Discussion by ali-0315** [vercel/community#988] [Help] [answered]
hi guys
how can i upload my media files in vercel?
i use django

**Comment by amyegan** [comment, marked as answer]
Hi @ali-0315! Can you walk me through what you mean? I'm guessing it's either media files for your site that should be uploaded for deployment, or media files that will be uploaded by users of your app. I don't want to give bad answers based on assumptions about what you're building. The more detail you can share, the better!

**Reply by ali-0315** [reply]
There is media files that will be uploaded by users of my website , when files uploaded i got a error that said you have not access to write
I need to upload some files dynamically while the project is running

**Reply by amyegan** [reply]
Okay, that makes sense. You won't be able to write files to your project from Serverless Functions or client-side code since the deployed copy is read-only. Since you need to upload files from users, a [file storage](https://vercel.com/docs/concepts/solutions/file-storage) or [database](https://vercel.com/docs/concepts/solutions/databases) solution would give you persistent storage.

**Reply by ali-0315** [reply]
Thanks 👍
